src/web/routes/application/children-dob/children-dob.js

```diff
@@ -20,20 +21,36 @@ const addActionRequested = body => body.hasOwnProperty('add')
 
 const extractChildrenEntries = (val, key) => key.startsWith('child')
 
-const behaviourForGet = (req, res, next) => {
+const initialiseChildrenInSession = (req) => {
   if (!req.session.hasOwnProperty('children')) {
-    req.session.children = {}
+    req.session.children = {
+      inputCount: 1,
+      // TODO HTBHF-1678 review if `childCount` is required for validation, else remove
+      childCount: 0
+    }
   }
 
-  req.session.children.count = 1
+  return req
+}
 
+const behaviourForGet = (req, res, next) => {
+  req = initialiseChildrenInSession(req)
   res.locals.children = req.session.children
   next()
 }
 
 const behaviourForPost = (req, res, next) => {
+  const childCount = countKeysContainingString('day', req.body)
```

Comment:
Just a thought: is it worth matching on `-day-` to prevent accidentally matching on a word that happens to contain `day`?  I know there is no such input at the moment.

Reply:
Good question. @mikeSmith08 any thoughts? Maybe we should be even more explicit and say `childDob-day-`

Reply:
.. actually the format is `childDob-1-day` so would have to be `-day`